src/main/kotlin/com/hj/leetcode/kotlin/problem17/Solution.kt

```diff
@@ -0,0 +1,51 @@
+package com.hj.leetcode.kotlin.problem17
+
+/**
+ * LeetCode page: [17. Letter Combinations of a Phone Number](https://leetcode.com/problems/letter-combinations-of-a-phone-number/);
+ */
+class Solution {
+
+    private val digitChars = mapOf(
+        '2' to "abc",
+        '3' to "def",
+        '4' to "ghi",
+        '5' to "jkl",
+        '6' to "mno",
+        '7' to "pqrs",
+        '8' to "tuv",
+        '9' to "wxyz"
+    )
+
+    /* Complexity:
+     * Time O(N * 4^N) and Space O(N * 4^N) where N is the length of digits;
+     */
+    fun letterCombinations(digits: String): List<String> {
+        if (digits.isEmpty()) {
+            return emptyList()
+        }
+
+        val result = mutableListOf<String>()
+        makeCombination(digits) { combination ->
+            result.add(combination)
+        }
+        return result
+    }
+
+    private fun makeCombination(
+        digits: String,
+        currentIndex: Int = 0,
+        currentLetters: StringBuilder = StringBuilder(),
```

Comment:
The `currentIndex` is a bit confusing, especially when there is `currentLetters`. Actually, we can get rid of `currentIndex` and use the length of `currentLetters`.

Reply:
Addressed at https://github.com/hj-core/hj-leetcode-kotlin/pull/117/commits/a7a4f44ad78a6f9b4a3b943ea530b80663e84cf3